ibis/expr/builders.py

```diff
@@ -66,7 +66,10 @@ def when(self, case_expr, result_expr):
         result_expr = rlz.any(result_expr)
 
         if not rlz.comparable(self.base, case_expr):
-            raise TypeError('Base expression and passed case are not ' 'comparable')
+            raise TypeError(
+                f'Base expression {rlz._arg_type_error_format(self.base)} and '
+                f'case {rlz._arg_type_error_format(case_expr)} are not comparable'
```

Comment:
The smallest of tests here might be a good idea so that we can ensure this doesn't regress.

Reply:
Agreed.  Pushed up.

Reply:
Thanks!